apps/ex_wire/lib/ex_wire/struct/bucket.ex

```diff
@@ -0,0 +1,273 @@
+defmodule ExWire.Struct.Bucket do
+  @moduledoc """
+  Represents a Kademlia k-bucket.
+  """
+
+  alias ExWire.Struct.{Peer, Bucket}
+  alias ExWire.Util.Timestamp
+  alias ExWire.KademliaConfig
+
+  defstruct [:id, :nodes, :updated_at]
+
+  @type t :: %__MODULE__{
+          id: integer(),
+          nodes: [Peer.t()],
+          updated_at: integer()
+        }
+
+  @doc """
+  Creates new bucket.
+
+  ## Examples
+      iex> ExWire.Struct.Bucket.new(1, time: :test)
+      %ExWire.Struct.Bucket{
+        id: 1,
+        nodes: [],
+        updated_at: 1525704921
+      }
+
+  """
+  @spec new(Peer.t()) :: t()
+  def new(id, options \\ [time: :actual]) do
+    %__MODULE__{
+      id: id,
+      nodes: [],
```

Comment:
I suggest using a map here instead since we'll need to lookup peers by their `remote_id`'s very often.